src/rai_bench/rai_bench/tool_calling_agent_bench/agent_bench.py

```diff
@@ -0,0 +1,211 @@
+
+loggers_type = logging.Logger
+
+
+class TaskResult(BaseModel):
+    task: str = Field(..., description="The task to be executed.")
+    model: str = Field(..., description="AI model used.")
+    success: bool = Field(
+        ..., description="Whether the task was successfully completed."
+    )
+    errors: List[str] = Field(
+        ..., description="List of errors that occurred during the task execution."
+    )
+    total_time: float = Field(..., description="Total time taken to complete the task.")
+    run_id: UUID = Field(..., description="UUID of the task run.")
+
+
+class BenchmarkSummary(BaseModel):
+    model: str = Field(..., description="AI model used.")
+    success_rate: float = Field(
+        ..., description="Percentage of successfully completed tasks."
+    )
+    avg_time: float = Field(..., description="Average time taken across all tasks.")
+    total_tasks: int = Field(..., description="Total number of tasks executed.")
+    run_id: UUID = Field(..., description="UUID of the average results entry.")
+
+
+class ToolCallingAgentBenchmark:
+    """
+    Benchmark for LangChain tool calling agents.
+    """
+
+    def __init__(
+        self,
+        tasks: Sequence[ToolCallingAgentTask],
+        logger: loggers_type | None = None,
+        results_filename: str = "agent_benchmark_results.csv",
+        summary_filename: str | None = None,
+    ) -> None:
+        self._tasks: Iterator[Tuple[int, ToolCallingAgentTask]] = enumerate(iter(tasks))
+        self.num_tasks = len(tasks)
+        self.task_results: List[TaskResult] = []
+        self.results_filename = results_filename
+        self.summary_filename = summary_filename or results_filename.replace(
+            ".csv", "_summary.csv"
+        )
+        self.fieldnames = [field for field in TaskResult.__annotations__.keys()]
+        self.summary_fieldnames = [
+            field for field in BenchmarkSummary.__annotations__.keys()
+        ]
+        self._initialize_results_file()
+        self._initialize_summary_file()
+        self.score_tracing_handler = ScoreTracingHandler()
+        self.model_results: Dict[str, List[TaskResult]] = {}
+        if logger:
+            self.logger = logger
+        else:
+            self.logger = logging.getLogger(__name__)
+
+    def run_next(self, agent, model_name: str) -> None:
+        try:
+            i, task = next(self._tasks)
+            self.logger.info(
+                f"RUNNING TASK NUMBER {i + 1} / {self.num_tasks}, TASK {task.get_prompt()}"
+            )
+            callbacks = self.score_tracing_handler.get_callbacks()
+
+            run_id = uuid.uuid4()
+            config: RunnableConfig = {
+                "run_id": run_id,
+                "callbacks": callbacks,
+                "tags": [task.complexity, model_name],
+                "recursion_limit": task.recursion_limit,
+            }
+            ts = time.perf_counter()
```

Comment:
Improve code readibility
```suggestion
            }
            
            ts = time.perf_counter()
```